addons/link_tracker/models/link_tracker.py

```diff
@@ -112,26 +113,60 @@ def _get_title_from_url(self, url):
 
         return title
 
-    @api.model
-    def create(self, vals):
-        create_vals = vals.copy()
+    @api.constrains('url', 'campaign_id', 'medium_id', 'source_id')
+    def _check_unicity(self):
+        """
+        Can not be implemented in SQL because we want to care about null values.
+        """
+        # build a query to fetch all needed link trackers at once
+        search_query = []
+        for tracker in self:
+            search_query = ['|'] + search_query + [
```

Comment:
maybe we could use expression.OR and .AND ?

Reply:
Good idea :) With expression.OR/AND I can even use list comprehension